src/screens/signIn/SignInScreen.tsx

```diff
@@ -1,44 +1,95 @@
-import React, {useState} from 'react'
-import {View} from 'react-native'
-import {Button, Input} from 'react-native-elements'
+import React, {FC, useEffect, useRef, useState} from 'react'
+import {
+  ActivityIndicator,
+  Animated,
+  Text,
+  TouchableHighlight,
+  View,
+} from 'react-native'
+import {TextInput} from '../../components/TextInput'
 import {
   sharedUseCreateTokenMutation,
   useCreateTokenMutation,
 } from '../../features/auth'
 import styles from './styles'
 
-export const SignInScreen = () => {
+export const SignInScreen: FC<any> = ({navigation}) => {
   // @TODO: remove placeholder later
   const [username, setUsername] = useState('maksim@example.com')
   const [password, setPassword] = useState('q12345')
-  const [createToken] = useCreateTokenMutation({
-    fixedCacheKey: sharedUseCreateTokenMutation,
-  })
+  const [startAnimatedValue, setAnimatedValue] = useState(0)
+  const animation = useRef(new Animated.Value(startAnimatedValue)).current
+  const [createToken, {error, data, isLoading, isSuccess, isError}] =
+    useCreateTokenMutation({
+      fixedCacheKey: sharedUseCreateTokenMutation,
+    })
+
+  useEffect(() => {
+    if (isError) {
+      Animated.timing(animation, {
+        toValue: 1,
+        duration: 500,
+        useNativeDriver: true,
+      }).start()
+    } else {
+      setAnimatedValue(0)
+    }
+  }, [isError, animation, error])
+
+  useEffect(() => {
+    // @ts-ignore
+    if (error?.status === 'FETCH_ERROR') {
+      navigation.navigate('NoInternetModal', {
```

Comment:
also you have this handling only here. So that other places where you are using network calls would not handle network connection?

Reply:
@Knight704 
A good point, I thought of this problem, but decided to leave as it.
I think it's worth creating a custom hook, which could accept an object with an error, and if the error is not empty navigate a user to appropriate Modal. 
The only question can I get `navigation` instance from this custom hook (whether we have possibility to call for example, inside hook like this 
```
const {navigation} = useNavigation();
```
or I need to pass the navigation instance to my custom hook directly?